main.go

```diff
@@ -64,11 +86,27 @@ func main() {
 		DdAppKey:   ddAppKey,
 	}
 
-	report, err := report.Generate(request)
+	content, err := report.Generate(generateRequest)
+
 	if err != nil {
 		exit("error generating report: %v", err)
+	} else if doUpload {
+		uploadRequest := report.UploadRequest{
+			ConfluenceSubdomain: *confSubdomain,
+			ConfluenceUsername:  confUsername,
+			ConfluenceToken:     confToken,
+			SpaceKey:            *spaceKey,
+			ParentId:            *parentId,
+			MarkdownContent:     content,
+		}
+		err = report.Upload(uploadRequest)
+		if err != nil {
+			exit("error uploading report: %v", err)
+		} else {
+			fmt.Println("Report uploaded successfully")
+		}
 	} else {
-		fmt.Println(report)
+		// If not uploading, just dump to stderr.
```

Comment:
Did you mean stdout?
```suggestion
		// If not uploading, just dump to stdout.
```

Reply:
Done